### PR TITLE
chore(web): KMW dev-tools reorganization, cleanup, script updates 📜

### PR DESCRIPTION
**Pull request by jahorton**
@keymanapp-test-bot skip

### CODE REVIEW

File: web/tools/build.sh
Comment (mcdurdin) [inline web/tools/build.sh:16]:
```suggestion
cd "$THIS_SCRIPT_PATH"
```

Given we have `$THIS_SCRIPT_PATH`, I'm trying to use this consistently in updates to build scripts, saves another call to `dirname` :grin:

File: web/tools/build.sh
Comment (mcdurdin) [inline web/tools/build.sh:25]: The following should work for restricting one dependency to `:recorder`:

```suggestion
builder_describe "Builds the Keyman Engine for Web's development & unit-testing tools" \
  "@../../common/web/keyman-version" \
  "@../../common/web/keyboard-processor" \
  "@../../common/web/recorder               :recorder" \
```

(Let me know if this doesn't work -- it's supposed to!)

File: web/tools/build.sh
Comment (mcdurdin) [inline web/tools/build.sh:29]: It'd be good for targets to match the folder names -- either use the target `:device-detect`, or use the folder name `:device`. Same pattern for `:wrappers`. Just reduces confusion and enables potential future automation.

File: web/tools/build.sh
Comment (mcdurdin) [inline web/tools/build.sh:39]: Conceptually, we could split this into three build.sh scripts, one for each target, and then have a parent script which just calls the three? Not sure if we need that level of structure at this point for tooling folders like this, but something to think about.

File: web/unit_tests/base.conf.js
Comment (mcdurdin) [inline web/unit_tests/base.conf.js:57]: We may include .woff files in the future, although .eot are thankfully dead. So perhaps we should leave this line in?

File: web/tools/device-detect/readme.md
Comment (mcdurdin) [inline web/tools/device-detect/readme.md:6]:
```suggestion
At this time, said code is included as part of KMW's main source. It is provided here as a 
separate module for unit tests.
```

File: web/tools/element-wrappers/readme.md
Comment (mcdurdin) [inline web/tools/element-wrappers/readme.md:6]:
```suggestion
At this time, said code is included as part of KMW's main source. It is provided here as a 
separate module for unit tests.
```

File: web/unit_tests/test.sh
Comment (mcdurdin) [inline web/unit_tests/test.sh:127]: Presumably you'll turn this into a dependency at some point when you update this script?

**Review by mcdurdin — APPROVED** [review]
LGTM. Just a few minor nits.

File: web/unit_tests/base.conf.js
Comment (jahorton) [reply web/unit_tests/base.conf.js:57]: When executing unit tests, we get a runtime warning about there being no match.  I figured I'd eliminate the warning.

File: web/unit_tests/test.sh
Comment (jahorton) [reply web/unit_tests/test.sh:127]: That's the plan for the next round, yes.

File: web/tools/build.sh
Comment (jahorton) [reply web/tools/build.sh:39]: I suppose we _could_; I just didn't see much of a point given how simple and straight-forward each one's build is.

Also note that I actually eliminated the previous one for `build:recorder` since it was legacy-style.

**Comment by jahorton** [comment]
Something of note that is possibly worth discussion - as I've started looking at the main `web/source/build.sh` script, I noticed that there's a `sourcemap-root` folder under `tools/` that's used for the main build, as opposed to the rest of the tools that are used for unit tests.

Feels kind of odd to leave it out of the general `tools/` script I'm writing here, but use of `sourcemap-root` is pretty much orthogonal to the other `tools/` subfolders, so... yeah.

**Comment by mcdurdin** [comment]
> Feels kind of odd to leave it out of the general `tools/` script I'm writing here, but use of `sourcemap-root` is pretty much orthogonal to the other `tools/` subfolders, so... yeah.

Yep. I anticipate that we may want to move files around for clearer organisation. This kind of thing is all pretty normal 😆.

File: web/tools/build.sh
Comment (mcdurdin) [reply web/tools/build.sh]: It's more about modularization and consistency than efficiency. If each module has its own build script, there's less for devs to remember. But I don't think we need to necessarily do this, just a consideration.

**Comment by jahorton** [comment]
> > Feels kind of odd to leave it out of the general `tools/` script I'm writing here, but use of `sourcemap-root` is pretty much orthogonal to the other `tools/` subfolders, so... yeah.
> 
> Yep. I anticipate that we may want to move files around for clearer organisation. This kind of thing is all pretty normal 😆.

Part of me wants to tackle that as part of this chain - the `.build-builder` file makes a good 'flag' of sorts for our CI.  But... that's the kind of thing we normally do early in a release cycle.  Should I do some reorg now anyway?

**Comment by mcdurdin** [comment]
> Should I do some reorg now anyway?

Yes, it's fine to do this now. I wouldn't do large-scale reorgs but this should be reasonable.

**Comment by jahorton** [comment]
> > Should I do some reorg now anyway?
> 
> Yes, it's fine to do this now. I wouldn't do large-scale reorgs but this should be reasonable.

This has been started as #7513 and #7515.

Any reason this particular PR should hold off until those two are ready?

**Comment by mcdurdin** [comment]
> Any reason this particular PR should hold off until those three are all ready?

Just re-running iOS build. I sure it's totally fine but it's neater anyway to have all-green. Once that passes, I think we are good to merge -- better now than in beta!

(With one agent down, there is a bit of delay in starting the build, but in next couple of hours this should ready to merge.)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.89-alpha](https://keyman.com/downloads/releases/alpha/16.0.89)

**Pull request merged**
